### PR TITLE
Fix external-to-internal link converter detection when using non-root path for `wagtail_serve` view

### LINKED ISSUE #11996: Rich text external-to-internal link converter not working when using a non-root path for `wagtail_serve`
### Issue Summary

Entering an Internal link in the External Link for Rich Text Editor isn't converting to an internal link.

### Steps to Reproduce

1. in settings.py enable Link Conversion by setting
-  WAGTAILADMIN_EXTERNAL_LINK_CONVERSION = "confirm"  
        or
-  WAGTAILADMIN_EXTERNAL_LINK_CONVERSION = "all"
2. Open a page in wagtail to edit
3. Add a paragraph tag, link some text using the link builder
image

I have tried a relative link and a fully qualified URL in my local, feature, and production environments, but the link isn't converting.

- /the-ascent/credit-cards/best-hotel-credit-cards/

**To get this functionality to work locally,** 
I had to hack the **.venv/lib/python3.11/site-packages/wagtail/admin/views/chooser.py**  file to remove the first instance of things in the components when doing a page match. 

``` python

                    components = [
                        component for component in url.split("/") if component
                    ]
                    if "localhost" in settings.WAGTAILADMIN_BASE_URL:
                        components = components[1:]
                    route = Site.objects.get(pk=pk).root_page.specific.route(
                        request,
                        components,
                    )
```

if the setting is ="confirm", 
I'd expect to get the popup:
image

If it's "all, I'd expect the next time I'd edit the link it would be automatically set to "Internal Link" and the page that it's currently set to as the highlighted option. **NOTE this is not what is highlighted**, its parent page or the parent page of the page I'm linking from is what is highlighted:
image

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (no)

### Technical details

-   Python version:3.11.1
-   Django version: 5.0.3
-   Wagtail version: 6.1.2
-   Browser version: Chrome on Mac 125.0.6422.113

### PR DESCRIPTION
**Pull request by laymonage**
Fixes #11996.

There's a report that this used to work at some point before 4.0, but I'm pretty sure this has always been the case ever since the feature was introduced in #7376.

To test:

- Create a Wagtail project that uses a non-root path when including `wagtail_urls`, e.g. `path("the-ascent/", include(wagtail_urls)),`
- Update the `HomePage` model to have a `RichTextField`/`RichTextBlock`. I'm using this example (which includes some leftover I have for testing #12002)
  ```py
  from wagtail.documents.blocks import DocumentChooserBlock
  from wagtail.documents.models import AbstractDocument, Document
  from wagtail.blocks import RichTextBlock
  from wagtail.admin.panels import FieldPanel
  from wagtail.fields import RichTextField, StreamField
  from wagtail.models import Page
  
  
  class HomePage(Page):
      body = StreamField(
          block_types=[
              ("doc", DocumentChooserBlock()),
              ("text", RichTextBlock()),
          ],
          blank=True,
      )
      rich = RichTextField(blank=True, null=True)
      content_panels = Page.content_panels + [FieldPanel("body"), FieldPanel("rich")]
  
  
  class CustomDocument(AbstractDocument):
      my = RichTextField()
      admin_form_fields = ("my",) + Document.admin_form_fields
  ```
- Update the Site instance in Settings > Sites to use localhost with the port you are using for the server e.g. 8000
- Create a few pages, e.g.
  - "Credit cards" `credit-cards/`
  - "Best hotel credit cards" `credit-cards/best-hotel-credit-cards/`
- Create an external link and enter `/the-ascent/credit-cards/best-hotel-credit-cards/`
  - Before this PR, this link would not be detected as an internal link and thus will become an external link
  - After this PR, this link would be converted to an internal link to the "Best hotel credit cards" page.
    - This may not be obvious, set `WAGTAILADMIN_EXTERNAL_LINK_CONVERSION = "confirm"` in settings to make it more explicit
- Create another external link and enter `/credit-cards/best-hotel-credit-cards/`
  - Before this PR, this relative link is detected as an internal link
  - After this PR, this relative is still detected as an internal link
- Repeat the above cases with the full URL instead of relative links, e.g. `http://127.0.0.1:8000/the-ascent/credit-cards/best-hotel-credit-cards/` and `http://127.0.0.1:8000/credit-cards/best-hotel-credit-cards/`.
- If necessary, add another site in Settings > Site and ensure that the conversion works with the new site (although relative URL conversion will not work once you have more than one site).

I made a repo for my test project in case you find it helpful: https://github.com/laymonage/wagtail-11996-repro

### CODE REVIEW

File: wagtail/admin/views/chooser.py [hunk 1]
```diff
@@ -642,40 +643,45 @@ def post(self, request):
...
+                site = Site.objects.select_related("root_page").get(pk=pk)
+                root_page = site.root_page.specific
+                for url in possible_urls:
+                    try:
+                        route = root_page.route(
+                            request,
+                            [component for component in url.split("/") if component],
+                        )
+                    except Http404:
+                        continue
```
Comment (laymonage) [inline wagtail/admin/views/chooser.py:684]: As far as I understand, `Http404` will only be raised by `route()` in this block, so I kept this simple by moving the `except Http404` all the way up here instead of encompassing the whole conversion logic. This also helps avoid unnecessary whitespace changes after I changed the structure to use a `dict[str, list]` which meant I had to do a nested loop here.

**Review by gasman — APPROVED** [review]
LGTM!
